ftdc/ftdc.go

```diff
@@ -415,6 +429,15 @@ func (ftdc *FTDC) getWriter() (io.Writer, error) {
 	// It's unclear in what circumstance we'd expect creating a new file to fail. Try 5 times for no
 	// good reason before giving up entirely and shutting down FTDC.
 	for numTries := 0; numTries < 5; numTries++ {
+		// The viam process is expected to be run as root. The FTDC directory must be readable by
+		// "other" users.
+		//
+		//nolint:gosec
+		if err = os.MkdirAll(ftdc.ftdcDir, 0o755); err != nil {
```

Comment:
Won't this recreate the FTDC directory on every attempt to create a single new FTDC file?

Reply:
MkdirAll has the same behavior as `mkdir -p`. Where it's a no-op + no error if the directory already exists.

FWIW, the case you're describing is exercised in a test.